Source/WebCore/html/parser/HTMLPreloadScanner.cpp

```diff
@@ -53,26 +53,26 @@ using namespace HTMLNames;
 
 TokenPreloadScanner::TagId TokenPreloadScanner::tagIdFor(const HTMLToken::DataVector& data)
 {
-    AtomString tagName(data);
-    if (tagName == imgTag)
+    auto knownTagName = AtomString::lookUp(data.data(), data.size());
```

Comment:
If this lookUp returns null, then, it means that this data is not a known one. So how about,

```
if (knownTagName.isNull())
    return TagId::Unknown;
```

Reply:
Sure, I’ll add that.

Reply:
Do we really need to optimize that case, though? I was just counting on the fact that null is not equal to the other tag names.

Reply:
New version uses `SortedArrayMap` here instead.